### ISSUE TITLE
Define status of keyIds in specified in initData passed to generateRequest()

### ISSUE THREAD

**Issue by ghost** [w3c/encrypted-media#130]
We should define what MediaKeyStatus we assign to keyIds which are specified in initData for the period from which they are known to the CDM until they become usable.

For example, say I go:

    navigator.requestMediaKeySystemAccess('org.w3.clearkey', [{initDataTypes: ['keyids']}]).then(
      (access) => access.createMediaKeys()
      ).then(
        (mediaKeys) => {
          var session = mediaKeys.createSession("temporary");
          var te = new TextEncoder();
          var initData = te.encode('{"kids":["LwVHf8JLtPrv2GUXFW2v_A"]}');
          return session.generateRequest('keyids', initData);
        }
      ));

What is the MediaKeyStatus of key specified in the "keyids" initData at the time when the generateRequest() promise has been resolved, but before the license has been passed to the CDM?

I propose:
* that we should assign MediaKeyStatus "status-pending" to keyIds specified in the initData passed to generateRequest().
* that we should run the "Update Key Statuses" algorithm at the end of generateRequest(). Note that this may cause "keystatuseschange" to fire.

For a text change, I propose we inject similar text to what's in the load() and update() algorithms into generateRequest() that is. I propose we insert after step 9 substep 10 in the generateRequest() definition:

>If the loaded session contains information about any keys (there are known keys), run the Update Key Statuses algorithm on the session, providing each key's key ID along with the appropriate MediaKeyStatus.

>Should additional processing be necessary to determine with certainty the status of a key, for example the key's ID was specified in the initData and a license must first be acquired for it to be usable, use "status-pending".

**Comment by ddorwin** [comment]
Currently, key IDs in the `initData` are not considered "known" to the session and thus not reflected in `keyStatuses`. They are not known and reflected until `update()` is called with a license that contains them. Before defining the status, we would need to decided to expose them before `update()`.

I think the main reasons are:
* There is no requirement that `initData` contains key IDs (or that the CDM can interpret them). For example, a content identifier may represent a list of keys for a movie.
* There is no requirement that `initData` contain all keys that will be returned in the license. (The license might even contain a subset of those requested.)

Note that the final proposed text refers to "the loaded session" as in the `load()` algorithm. For `generateRequest()`, all key IDs in the `initData` would be ` "status-pending"`.

**Comment by mwatson2** [comment]
David is correct that at present keys do not become known until they key itself is provided via `update()`. If we introduce a mechanism to provide keys themselves through `initData`, for example under #132, then we will need the changes proposed here too. In the meantime, I suggest we close this separate issue.

**Comment by ddorwin** [comment]
Yes, if another mechanism for providing keys is added, the relevant algorithm(s) would need to call the [Update Key Statuses](https://w3c.github.io/encrypted-media/#update-key-statuses) algorithm.

I agree with closing this issue.
